src/main/java/com/aws/greengrass/smbridge/clients/MQTTClient.java

```diff
@@ -81,79 +82,63 @@ public void deliveryComplete(IMqttDeliveryToken token) {
      *
      * @param topics             topics passed in by Nucleus
      * @param mqttClientKeyStore KeyStore for MQTT Client
+     * @param executorService    Executor service
      * @throws MQTTClientException if unable to create client for the mqtt broker
      */
     @Inject
-    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore) throws MQTTClientException {
-        this(topics, mqttClientKeyStore, null);
+    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService)
+            throws MQTTClientException {
+        this(topics, mqttClientKeyStore, executorService, null);
         // TODO: Handle the case when serverUri is modified
         try {
             this.mqttClientInternal = new MqttClient(serverUri, clientId, dataStore);
         } catch (MqttException e) {
-            throw new MQTTClientException("Unable to create a MQTT client", e);
+            throw new MQTTClientException("Unable to create an MQTT client", e);
         }
     }
 
-    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, MqttClient mqttClient) {
+    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService,
+                         IMqttClient mqttClient) {
         this.mqttClientInternal = mqttClient;
         this.dataStore = new MemoryPersistence();
         this.serverUri = Coerce.toString(topics.findOrDefault(DEFAULT_BROKER_URI,
                 KernelConfigResolver.CONFIGURATION_CONFIG_KEY, BROKER_URI_KEY));
-        this.clientId = Coerce.toString(topics.findOrDefault(SMBridge.SERVICE_NAME,
+        this.clientId = Coerce.toString(topics.findOrDefault(DEFAULT_CLIENT_ID,
                 KernelConfigResolver.CONFIGURATION_CONFIG_KEY, CLIENT_ID_KEY));
         this.mqttClientKeyStore = mqttClientKeyStore;
         this.mqttClientKeyStore.listenToUpdates(this::reset);
+        this.executorService = executorService;
     }
 
-    private void reset() {
+    void reset() {
         if (mqttClientInternal.isConnected()) {
             try {
                 mqttClientInternal.disconnect();
             } catch (MqttException e) {
-                LOGGER.atError().setCause(e).log("Failed to disconnect MQTT Client");
+                LOGGER.atError().setCause(e).log("Failed to disconnect MQTT client");
                 return;
             }
         }
 
         try {
-            readKeyStoreAndConnect();
-        } catch (KeyStoreException | MqttException e) {
-            LOGGER.atError().setCause(e).log("Failed to connect with updated KeyStore");
-            return;
+            connectAndSubscribe();
+        } catch (KeyStoreException e) {
+            throw new RuntimeException(e);
```

Comment:
Don't throw a RuntimeException here. It's too generic. Also logging is missing

Reply:
Do you think I should extend RuntimeException in my own implementation, or should I call some other exception that I already have implemented (MQTTClientException?)? Could also pass KeyStoreException up the call stack...

Reply:
Added logging.

Reply:
You won't be able to throw a checked exception here. I think, that is why someone chose to throw a Runtime exception. Since it is a runtime exception, I am not sure if we will get any value with wrapping it in our own impl